### DISCUSSION TITLE
Oobabooga text-generation-webui v1.15 stable is out!?

### DISCUSSION THREAD

**Discussion by Sonic2kDBS** [oobabooga/text-generation-webui#6435] [General]
Hi folks, I use Oobabooga text-gereation-webui for quite some time now. I think it is the best UI, you can have. The extensions are great and you can use it as API point and most important, you can have a lot of fun with it.

However, today I updated the web UI to version 1.15 and again something broke for the xth time and have to fully reinstall everything. For me it is not only over 20GB of data to download but also a copy session of all files back to their place and a renew of the configuration and settings. And of course time. I found this, what described my bug this time best: https://github.com/oobabooga/text-generation-webui/issues/6389 but I could see no replies, just frustration and I thought by myself, how much have left this great webui because of that until now. There would have been many replays some months ago.

I have to make a completely new setup now for the 8th time. and I have 3 running installations for different purposes, so this will be my 24th installation, if I am finished. Did I said I am still thankful, that development continues? If not, I am. But well I see a big problem here. And I think, we need a Stable Version soon. One, that runs and can be upgradet to the next stable version, without breaking In pieces. I still will have my testing setup but dude, give us something sable for inferencing please :)

I  hope, you like and support my idea. Have a great Weekend.
Ragnar / Sonic2k

**Comment by clover1980** [comment]
I'm still waiting when ability of using veeeery long chat history returns, but it was broken after December 2023 update, broken still.
I would advice to back up all OS image before any update, because you could lose something what was working previously, i can't install old version today because installer takes only the fresh requirements.

**Reply by Sonic2kDBS** [reply]
Yes, exactly. I also have backups and old installers. Can recommend that too. But that doesn't help for long, because from time to time an Update is needed anyway (for example for newer models). So a stable branch would be really nice to have.

**Reply by Sonic2kDBS** [reply]
Also, probably it is more easy to add patches then, like one that brings back the function "of using veeeery long chat history", like you need. It would work for six months at least. Until the next stable version would come out.

**Reply by clover1980** [reply]
I'm not sure it will be possible, they've changed tool architecture a lot. Kinda they provide today longer chats by "streaming llm" option, i've tested it, you can talk beyond model limit, but with every new prompt it removes an older phrase from the history, so kinda making free space by deleting.

**Comment by Sonic2kDBS** [comment]
I think, that this is a big problem. Probably creating a stable branch, that gets merged every 3 or 6 months with the main branch could solve this issue. The only thing to do is testing the update once every six months if anything breaks and make sure everything works. I think that is not that kind of a big deal. But it would be less of a headache of either try to fix issues or reinstall everything. Backup doesn't always work, if the environment is messed up. Installing with `git clone -b stable https://github.com/oobabooga/text-generation-webui.git` would be so nice. I hope this will become in consideration one time.

**Comment by Sonic2kDBS** [comment]
I have seen, that between the main and dev branch two new temporary branches did pop up.  Those are dependabot/pip/dev/accelerate-eq-1.0.star and dependabot/pip/dev/numpy-eq-2.1.star. That is great. It makes the main version like it is, while testing stuff. That is great.
But anyways, it would be nice to get some statement. Some response, some thoughts or ideas. Is the main branch the stable branch now? Or will we get a separate stable branch soon? Are inventions handled in different test branches now? Can we please have some clarity?
However, have a great weekend ya all :)

**Comment by Sonic2kDBS** [comment]
_r/Oobabooga • 11 days ago_
_Official subreddit for oobabooga/text-generation-webui, a Gradio web UI for Large Language Models._
_NotMyPornAKA_

**Why have all my models slowly started to error out and fail to load? Over the course of a few months, each one eventually fails without me making any modifications other than updating Ooba**

https://www.reddit.com/r/Oobabooga/comments/1g5h7p7/why_have_all_my_models_slowly_started_to_error/

**Reply by Sonic2kDBS** [reply]
I know Text generation web UI's goal is to become the AUTOMATIC1111/stable-diffusion-webui of text generation.
That is a great goal. Please keep on to reach it.
I installed my stable-diffusion-webui at 2023-06-07.
And I update it by a script with every start. It still works. It still loads all the models.
I want that for Oobabooga text-generation-webui too.
